pkg/cli/verifier/envoy_config.go

```diff
@@ -478,8 +480,28 @@ func (v *EnvoyConfigVerifier) getDstMeshServicesForSvc(svc corev1.Service) ([]se
 		// The endpoints for the kubernetes service carry information that allows
 		// us to retrieve the TargetPort for the MeshService.
 		meshSvc.TargetPort = k8s.GetTargetPortFromEndpoints(portSpec.Name, *endpoints)
-		meshServices = append(meshServices, meshSvc)
+
+		if !k8s.IsHeadlessService(svc) {
+			meshServices = append(meshServices, meshSvc)
```

Comment:
Should there be a `continue` after this to match pkg/k8s?
https://github.com/openservicemesh/osm/blob/b3d0c7ff9b02b8253c96f142415709544cbb1fc2/pkg/k8s/client.go#L368-L371

Reply:
Yes good catch